Assets/Scripts/Gamestate/PlayerManager.cs

```diff
@@ -227,10 +230,10 @@ public void SetLayer(int playerIndex)
         // Set layers for the camera to ignore (the other players' gun layers, and this layer)
         // Bitwise negation of this player's model layer and all gun layers that do not belong to this player
         // Gun layers are 4 above their respective player layers.
-        var playerAndGunMask = ((1 << playerLayer) | ((1 << (playerLayer + 4)) ^ allGunsMask));
+        var playerAndGunMask = ((1 << playerLayer) | allGunsMask) & ~(1 << (playerLayer + 4));
```

Comment:
Aha, guess I didn't test this enough to see the buggerino -_-
... though I'm not sure what (if anything) was wrong here, care to elaborate? The gun mask should become [bits in gun mask] except [player's gun bit] with the player gun XOR all guns expression 🤔

Reply:
It is the same, I changed it while testing and forgot about it 😅

Since you wrote the original, I'll let you decide which version we should keep (whatever we feel is most readable I guess)

Reply:
I do of course feel like my own version makes the most intuitive sense, but regardless of which on we go for I do think we should extract the bitshifting expressions to their own variables so that the whole expression becomes easier to read.

Reply:
Responded in another comment